torch/csrc/jit/passes/mkldnn_rewrite.cpp

```diff
@@ -0,0 +1,276 @@
+#include <ATen/Config.h>
+#include <ATen/code_template.h>
+#include <ATen/core/jit_type.h>
+#include <ATen/native/ConvUtils.h>
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/subgraph_matcher.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
+#include <torch/csrc/jit/passes/fold_conv_bn.h>
+#include <torch/csrc/jit/passes/freeze_module.h>
+#include <torch/csrc/jit/passes/graph_rewrite_helper.h>
+#include <torch/csrc/jit/passes/inliner.h>
+#include <torch/csrc/jit/passes/mkldnn_rewrite.h>
+#include <torch/csrc/jit/passes/remove_dropout.h>
+#include <torch/csrc/jit/runtime/graph_executor_impl.h>
+#include <torch/csrc/jit/tensorexpr/kernel.h>
+
+namespace torch {
+namespace jit {
+
+namespace mkldnn {
+namespace detail {
+
+#if AT_MKLDNN_ENABLED()
+bool mkldnn_fuser_enabled = true;
+#else
+bool mkldnn_fuser_enabled = false;
+#endif // AT_MKLDNN_ENABLED()
+
+} // namespace detail
+} // namespace mkldnn
+
+#if AT_MKLDNN_ENABLED()
+
+c10::VaryingShape<int64_t> getSizesOf(Node* n, size_t idx) {
+  auto tt = n->input(idx)->type()->cast<TensorType>();
+  return tt->sizes();
+}
+
+bool isContiguous(Value* v) {
+  auto const& tt = v->type()->cast<TensorType>();
+  if (!tt) {
+    return false;
+  }
+  if (!tt->isComplete()) {
+    return false;
+  }
+  auto const& sizes = tt->sizes().concrete_sizes();
+  auto const& strides = tt->strides().concrete_sizes();
+  if (!sizes || !strides) {
+    return false;
+  }
+  int ndims = (*sizes).size();
+  // TODO: only support 4D input for now
+  if (ndims != 4) {
+    return false;
+  }
+
+  // TODO: currently only support contiguous input. Channels last contiguous
+  // will be supported as a next step
+  return *strides == TensorType::contiguousStridesOf(*sizes);
+}
```

Comment:
This function already exists here, let's reuse it (please feel free to move it to another file if that helps):
https://github.com/pytorch/pytorch/blob/c7e8de6e86c32b727a6dc05bee645896ccc5262e/torch/csrc/jit/tensorexpr/kernel.cpp#L211-L235

Reply:
Modified the code to reuse this function: https://github.com/pytorch/pytorch/pull/77157/commits/0fc328b519648bd91ef50b274810cd7da3e6e799